llama_stack/apis/datasets/datasets.py

```diff
@@ -4,19 +4,96 @@
 # This source code is licensed under the terms described in the LICENSE file in
 # the root directory of this source tree.
 
-from typing import Any, Dict, List, Literal, Optional, Protocol
+from enum import Enum
+from typing import Annotated, Any, Dict, List, Literal, Optional, Protocol, Union
 
 from pydantic import BaseModel, Field
 
-from llama_stack.apis.common.content_types import URL
-from llama_stack.apis.common.type_system import ParamType
 from llama_stack.apis.resource import Resource, ResourceType
-from llama_stack.schema_utils import json_schema_type, webmethod
+from llama_stack.schema_utils import json_schema_type, register_schema, webmethod
+
+
+class DatasetPurpose(Enum):
+    """
+    Purpose of the dataset. Each type has a different column format.
+    :cvar post-training/messages: The dataset contains messages used for post-training. Examples:
+        {
+            "messages": [
+                {"role": "user", "content": "Hello, world!"},
+                {"role": "assistant", "content": "Hello, world!"},
+            ]
+        }
+    """
+
+    post_training_messages = "post-training/messages"
+    eval_question_answer = "eval/question-answer"
+
+    # TODO: add more schemas here
+
+
+class DatasetType(Enum):
+    """
+    Type of the dataset source.
+    :cvar huggingface: The dataset is stored in Huggingface.
+    :cvar uri: The dataset can be obtained from a URI. 
+    :cvar rows: The dataset is stored in rows. 
+    """
+    huggingface = "huggingface"
+    uri = "uri"
+    rows = "rows"
+
+
+@json_schema_type
+class URIDataSource(BaseModel):
+    """A dataset that can be obtained from a URI.
+    :param uri: The dataset can be obtained from a URI. E.g.
+        - "https://mywebsite.com/mydata.jsonl"
+        - "lsfs://mydata.jsonl"
+        - "data:csv;base64,{base64_content}"
+    """
+    type: Literal["uri"] = "uri"
+    uri: str
+
+
+@json_schema_type
+class HuggingfaceDataSource(BaseModel):
+    """A dataset stored in Huggingface.
+    :param path: The path to the dataset in Huggingface. E.g.
+        - "llamastack/simpleqa"
+    :param params: The parameters for the dataset.
+    """
+    type: Literal["huggingface"] = "huggingface"
+    path: str
+    params: Dict[str, Any]
+
+
+@json_schema_type
```

Comment:
I think these types are unlikely to be reused across the APIs (they will only be used in the dataset creation API) -- so not registering them with a `json_schema_type` keeps the OpenAPI spec minimal (those fields get inlined)

Reply:
Make sense, I have this habit of adding `@json_schema_type` everywhere.

Reply:
Wait, I assume you are referring to the Enums, and not DataSource?

I think DataSource type is still useful. 
<img width="1286" alt="image" src="https://github.com/user-attachments/assets/1600d8bd-25e0-4a41-ba84-8b752d6603ad" />